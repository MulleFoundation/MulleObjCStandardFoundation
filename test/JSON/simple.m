//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>


static void   code_decode( id value)
{
   NSData   *data;
   id       decoded;

   @try
   {
      data   = [NSPropertyListSerialization dataFromPropertyList:value
                                                          format:MullePropertyListJSONFormat
                                                errorDescription:NULL];
      decoded = [NSPropertyListSerialization propertyListFromData:data
                                                 mutabilityOption:0
                                                           format:NULL
                                                 errorDescription:NULL];
      printf( "%s->%s\n",
            [[value description] UTF8String],
            [[decoded description] UTF8String]);
   }
   @catch( NSException *exception)
   {
      printf( "exception: %s\n", [[exception reason] UTF8String]);
   }
}


int   main( int argc, char *argv[])
{
   code_decode( [NSNumber numberWithInt:1848]);
   code_decode( [NSNumber numberWithDouble:18.0e-48]);

   code_decode( @"");
   code_decode( @"bar");
   code_decode( @"/");
   code_decode( @"a space");

   return( 0);
}
