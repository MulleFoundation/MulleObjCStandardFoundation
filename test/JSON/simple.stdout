1848->1848
1.8e-47->1.8e-47
->
bar->bar
/->/
a space->a space
